### PR TITLE
Getting Completed and Expired Records.

### PR DESCRIPTION
**Pull request by mpenumal**
**Taiga User Story** US 650

**Taiga Task(s)** 652

**What did you change?** 
Additional check to get expired records along with completed records.

**How can we test?** 
Expired survey instance records should should be plotted along with the Completed records on the Patient Compliance Chart.

### CODE REVIEW

File: controller/helper/process-survey-instances.js [hunk 1]
```diff
@@ -102,14 +102,22 @@ function calculateTimeLeft (openTime, endTime, completedTime) {
...
+        // caculate percent of time taken out of total time availible to take the survey
+        percentTimeLeft = Math.round(timeTaken / totalAvailibleTime * percent);
+
+        // either take the amount of time left
+        // or if the survey instance expired (negative percent) show zero time left
+        return Math.max(percentTimeLeft, minTime);
+    }
+
+    // if the survey instance expired (NaN percent) show zero time left
+    return minTime;
```
Comment (ChristianMurphy) [inline controller/helper/process-survey-instances.js]: I think the return on line 116 can be moved here instead. (so there is only one return)

File: controller/helper/process-survey-instances.js [hunk 2]
```diff
@@ -102,14 +102,22 @@ function calculateTimeLeft (openTime, endTime, completedTime) {
...
     // calculate the time in hours until end time
     const totalAvailibleTime = endTime.diff(openTime, 'hours');
-    const timeTaken = endTime.diff(completedTime, 'hours');
+    let percentTimeLeft = minTime;
+    let timeTaken = minTime;
 
-    // caculate percent of time taken out of total time availible to take the survey
-    const percentTimeLeft = Math.round(timeTaken / totalAvailibleTime * percent);
+    if (completedTime !== null && !isNaN(completedTime)) {
+        timeTaken = endTime.diff(completedTime, 'hours');
```
Comment (ChristianMurphy) [inline controller/helper/process-survey-instances.js]: remove `let timeTaken = minTime;` from line 106, and make this line (109) `const`.

**Comment by poojaRal** [comment]
Once the code review comments are implemented rest looks good to me.
LGTM

File: controller/helper/process-survey-instances.js (hunk 2 above)
Comment (mpenumal) [inline controller/helper/process-survey-instances.js]: Done

File: controller/helper/process-survey-instances.js (hunk 1 above)
Comment (mpenumal) [inline controller/helper/process-survey-instances.js]: Done

**Comment by ChristianMurphy** [comment]
LGTM

**Pull request merged**